lib/core.ps1

```diff
@@ -351,15 +351,32 @@ function url_remote_filename($url) {
     return $basename
 }
 
-function ensure($dir) { if(!(test-path $dir)) { mkdir $dir > $null }; resolve-path $dir }
+function ensure($dir) { if (!(test-path $dir)) { mkdir $dir > $null }; resolve-path $dir }
+
 function Get-AbsolutePath {
-    param (
-        [Parameter(Mandatory = $true)]
+    <#
+        .SYNOPSIS
+            Get absolute path
+        .DESCRIPTION
+            Get absolute path, even if not existed
+        .PARAMETER Path
+            Path to manipulate
+        .OUTPUTS
+            System.String
+                Absolute path, may or maynot existed
+    #>
+    [CmdletBinding()]
+    [OutputType([String])]
+    Param (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
         [String]
         $Path
-    ) # should be ~ rooted
-    return $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($Path)
+    )
+    Process {
```

Comment:
What is advantage of having process for one line body function?

Reply:
Just for pipeline. Indeed, there're only three usages in scoop now, and if not used in manifest's script, there is no need to support pipeline.